jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/ServletChannel.java

```diff
@@ -781,14 +782,27 @@ void onTrailers(HttpFields trailers)
 
     public void onCompleted()
     {
-        HttpServletRequest httpServletRequest = _request.getHttpServletRequest();
+        ServletContextRequest.ServletApiRequest apiRequest = _request.getServletApiRequest();
         if (LOG.isDebugEnabled())
-            LOG.debug("onCompleted for {} written={}", httpServletRequest.getRequestURI(), getBytesWritten());
+            LOG.debug("onCompleted for {} written={}", apiRequest.getRequestURI(), getBytesWritten());
 
         long idleTO = _configuration.getIdleTimeout();
         if (idleTO >= 0 && getIdleTimeout() != _oldIdleTimeout)
             setIdleTimeout(_oldIdleTimeout);
 
+        if (getServer().getRequestLog() != null)
+        {
+            Authentication authentication = apiRequest.getAuthentication();
+            if (authentication instanceof Authentication.User userAuthentication)
+                _request.setAttribute(CustomRequestLog.USER_NAME, userAuthentication.getUserIdentity().getUserPrincipal().getName());
```

Comment:
Ultimately, it might be best to have an `EnumMap<CustomRequestLog.ATTR,String>` as part of the request, as a more efficient way to pass these items for requests that don't use the attributes.

This probably needs to be tested... but if we do decide that just using attributes is best/simplest, then perhaps we give up on the whole lazy creation of the attributes map underneath as needless complexity.

i.e. either we continue to strive to avoid routine request attributes or we remove lazy request attribute map.  It is silly to be lazy and routine.